### DISCUSSION TITLE
Need help regarding conditional flow rendering

### DISCUSSION THREAD

**Discussion by Xiddiqui** [tjtanjin/react-chatbotify#66] [General]
const flow = {
        start: {
            message: "Hello! How can I assist you today?",
            options: ["Add Client KYC"],
            path: (input) => {
                console.log("input", input);
                if (input.userInput === "Add Client KYC") {
                    return "form_fill";
                }
                return "loop";
            },
        },
        loop: {
            message: async (params) => {
                const result = await fetchData(params.userInput);
                console.log(result);
                return result;
            },
            path: "loop",
        },
        form_fill: {
            message: "Please Fill the form bellow",
            render: <ChatKycForm onSubmit={addLead} />,
            chatDisabled: true,
        },
        end: {
            message: "Your Lead created",
            options: ["Add another Client KYC"],
            path: "form_fill",
        },
    };
    This is my flow function in form_fill when addLead api success it should go to my end block . 
    end: {
            message: "Your Lead created",
            options: ["Add another Client KYC"],
            path: "form_fill",
        },
        How can i achieve this .

**Comment by tjtanjin** [comment]
Hey @Xiddiqui, both the `path` and `render` attributes are able to take in a function. If you want to conditionally determine what to render or where to path to, you can use a function. The attributes are documented here: https://react-chatbotify.tjtanjin.com/docs/api/attributes

Let me know if you need more help!

**Reply by Xiddiqui** [reply]
Thank you @tjtanjin  but i need little more help . 
In render i am showing a form of three text field and one button on pressing submit i am calling a function of addLead which is out side of my flow function now i want on success of my api user receive message of success . In my flow function i have added end {..} but how can i set on api success

**Reply by tjtanjin** [reply]
Hey @Xiddiqui! If I am understanding you correctly, you are trying to navigate between paths with logic outside of the flow. If that is the case, you can use the advance custom paths option that is shown in the example here: https://react-chatbotify.tjtanjin.com/docs/examples/custom_paths

If this is not what you're looking for, feel free to share your code for the `render` portion and I'll try to help 😊

**Reply by Xiddiqui** [reply]
Thank you @tjtanjin . My issue solved by following this link . Thanks again

**Reply by tjtanjin** [reply]
No problem, feel free to reach out if you have more issues 😊
